src/app/shared/entry/url-deconstruct.pipe.ts

```diff
@@ -0,0 +1,17 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@ Pipe({
+  name: 'urlDeconstruct'
+})
+export class UrlDeconstructPipe implements PipeTransform {
+
+  transform(providerUrl: string, versionName: string): any {
+    if (providerUrl.includes('github.com') && versionName !== '') {
+      return (new URL(providerUrl).pathname).substr(1) + '/tree/' + versionName;
```

Comment:
You have `new URL(providerUrl).pathname).substr(1)` 3 times...

You could just do it once, at the top of function, and then add the bitbucket or github path in your if conditions.